sunspec.go

```diff
@@ -28,10 +28,125 @@ type ScaleFactor int16
 
 type Count uint16
 
+// An array is a collection of devices.
+//
+// This type is used when the underlying representation
+// is an XML document.
```

Comment:
It is - the motivation for the Array type is the fact that the XML format allows multiple devices to be described by one document.